### ISSUE TITLE
boundary event coordinates do not cover bottom of visible map

### ISSUE THREAD

**Issue by mrdvt92** [dceejay/RedMap#210]
node-red 1.06
worldmap 2.30.1 (issues was in 2.29.1 too)

I was noticing that my map did not have any points at the bottom and so I drew the bbox on the map and there is a gap at the bottom.  But, I expect  the bounding box to cover the entire map extents.

![image](https://user-images.githubusercontent.com/5374501/198093404-b70ac239-2edf-4ce6-b8cb-8963617ee473.png)

This is my map configuration but enabling/disabling coordinates display does not change behavior

![image](https://user-images.githubusercontent.com/5374501/198093909-074d7b55-e611-4679-b5b4-a1c3b24f8db0.png)

This is the template node to make the bbox on change of boundary.

```
[{"id":"9b2015b6.65a058","type":"worldmap in","z":"e02f29e1.67d3d8","name":"worldmap boundary","path":"/worldmap","events":"bounds","x":110,"y":180,"wires":[["7c761ea0.1480d","9470a517.f4e368","aa3569bc.33e3a8"]]},{"id":"aa3569bc.33e3a8","type":"trigger","z":"e02f29e1.67d3d8","op1":"","op2":"","op1type":"nul","op2type":"pay","duration":"250","extend":true,"units":"ms","reset":"","bytopic":"all","name":"","x":420,"y":180,"wires":[["3a7daa20.4a5456"]]},{"id":"3a7daa20.4a5456","type":"change","z":"e02f29e1.67d3d8","name":"set center","rules":[{"t":"set","p":"payload.center","pt":"msg","to":"{}","tot":"json"},{"t":"set","p":"payload.center.lat","pt":"msg","to":"$average([payload.north, payload.south])\t","tot":"jsonata"},{"t":"set","p":"payload.center.lon","pt":"msg","to":"$average([payload.east, payload.west])\t","tot":"jsonata"}],"action":"","property":"","from":"","to":"","reg":false,"x":580,"y":180,"wires":[["495f29b2.e5e508","9ed54ec.ce141b","c28055e3.2c3cf8"]]},{"id":"c28055e3.2c3cf8","type":"template","z":"e02f29e1.67d3d8","name":"","field":"payload","fieldType":"msg","format":"handlebars","syntax":"mustache","template":"{\n  \"name\": \"bounds\", \n  \"area\": [\n            [{{payload.south}}, {{payload.west}}], \n            [{{payload.north}}, {{payload.west}}], \n            [{{payload.north}}, {{payload.east}}],\n            [{{payload.south}}, {{payload.east}}],\n            [{{payload.south}}, {{payload.west}}]\n    ]}","output":"json","x":1060,"y":400,"wires":[["c5157a30.bbb588","4c9dc44a.908fdc"]]},{"id":"4c9dc44a.908fdc","type":"ui_worldmap","z":"e02f29e1.67d3d8","group":"b216f4f.7d69c08","order":1,"width":"20","height":"10","name":"","lat":"38.2","lon":"-97.2","zoom":"4.5","layer":"","cluster":"","maxage":"","usermenu":"hide","layers":"show","panit":"false","panlock":"false","zoomlock":"false","hiderightclick":"true","coords":"none","showgrid":"false","allowFileDrop":"false","path":"/worldmap","overlist":"","maplist":"OSMG,OSMC,EsriC,EsriS,EsriT,EsriDG","mapname":"","mapurl":"","mapopt":"","mapwms":false,"x":1680,"y":200,"wires":[]},{"id":"b216f4f.7d69c08","type":"ui_group","z":"","name":"World Map","tab":"a4e26d94.4446","order":1,"disp":true,"width":"20","collapse":false},{"id":"a4e26d94.4446","type":"ui_tab","z":"","name":"World Map","icon":"dashboard","order":6,"disabled":false,"hidden":false}]
```

**Comment by dceejay** [comment]
The bounds event is reported after a moveend or zoomend event within leaflet. The co-ordinates are those given by map.getBounds()... so not sure why they would be wrong - but likewise not sure if it's worldmap or the underlying leaflet.

**Comment by mrdvt92** [comment]
> not sure if it's worldmap or the underlying leaflet.

I added this leaflet example to this same node implementation and it reports the south/west corner correctly.  In this example the click event is registered to send a message with the click coordinates and the bbox.

```
[{"id":"5c525d5a.2d40a4","type":"ui_template","z":"89b966af.00ed98","group":"63fc21fa.6e84e","name":"Leaflet","order":1,"width":"20","height":"10","format":"<link rel=\"stylesheet\" href=\"https://unpkg.com/leaflet@1.8.0/dist/leaflet.css\" integrity=\"sha512-hoalWLoI8r4UszCkZ5kL8vayOGVae1oxXe/2A4AO6J9+580uKHDO3JdHb7NzwwzK5xr/Fs0W40kiNHxM9vyTtQ==\" crossorigin=\"\"/>\n<script src=\"https://unpkg.com/leaflet@1.8.0/dist/leaflet.js\" integrity=\"sha512-BB3hKbKWOc9Ez/TAwyWxNXeoV9c1v6FIeYiBieIWkpLjauysF18NzgR1MBNBXf8/KABdlkX68nAhlwcDFLGPCQ==\" crossorigin=\"\"></script>\n\n<script>\n    (function() {\n        const cScope = this.scope;\n        setTimeout(function () {\n            \n        \tconst map = L.map('mapid').setView([38.780277, -77.386698], 13);\n            let marker;\n            \n        \tL.tileLayer('https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {\n               maxZoom: 19,\n               attribution: '&copy; <a href=\"http://www.openstreetmap.org/copyright\">OpenStreetMap</a>'\n        \t}).addTo(map);\n        \t\n        \tmap.on('click', function(e) {\n                cScope.send({\"action\":\"click\", \"point\":e.latlng, \"bbox\":map.getBounds()});\n                if (marker) {\n                    map.removeLayer(marker);\n                };\n                marker = L.marker(e.latlng).addTo(map);\n            });\n            \n            setTimeout(function () {\n                map.invalidateSize();\n            }, 100);\n        }, 300);\n    })();\n</script>\n\n<div id=\"mapid\" style=\"width: 100%; height: 100%; z-index: 1;\"></div>","storeOutMessages":true,"fwdInMessages":true,"resendOnRefresh":false,"templateScope":"local","x":90,"y":140,"wires":[["9bc1fa57.e63918","1f38776f.36a619","3626ac11.e223e4"]]},{"id":"3626ac11.e223e4","type":"debug","z":"89b966af.00ed98","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","x":790,"y":140,"wires":[]},{"id":"63fc21fa.6e84e","type":"ui_group","z":"","name":"OpenStreetMap Leaflet","tab":"61be5b66.568104","order":2,"disp":true,"width":"20","collapse":false},{"id":"61be5b66.568104","type":"ui_tab","z":"","name":"Map Example","icon":"dashboard","order":7,"disabled":false,"hidden":false}]
```

**Comment by dceejay** [comment]
If I use your original example it works fine...
```
[{"id":"c9828117.f463","type":"worldmap","z":"27d4cd61f8c2d5ed","name":"worldmap-/test","lat":"","lon":"","zoom":"","layer":"","cluster":"","maxage":"","usermenu":"show","layers":"show","panit":"false","panlock":"false","zoomlock":"false","hiderightclick":"false","coords":"none","showgrid":"false","allowFileDrop":"false","path":"/worldmap","overlist":"DR,CO,RA,DN,HM","maplist":"OSMG","mapname":"","mapurl":"","mapopt":"","mapwms":false,"x":455,"y":450,"wires":[]},{"id":"73c8a005.6a324","type":"worldmap in","z":"27d4cd61f8c2d5ed","name":"worldmap-/test","path":"/worldmap","events":"bounds","x":235,"y":510,"wires":[["d8b66827.f4d188","61b068dd229b63ee"]]},{"id":"8a82b6f9.b0c6c8","type":"inject","z":"27d4cd61f8c2d5ed","name":"add point to layer","repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"{\"lat\":38.1,\"lon\":-77.1,\"name\":\"myname\",\"layer\":\"mylayer\"}","payloadType":"json","x":255,"y":430,"wires":[["c9828117.f463"]]},{"id":"d8b66827.f4d188","type":"debug","z":"27d4cd61f8c2d5ed","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","x":425,"y":510,"wires":[]},{"id":"a7059aab.b14288","type":"inject","z":"27d4cd61f8c2d5ed","name":"clear","repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"{\"command\":{\"clear\":\"mylayer\"}}","payloadType":"json","x":225,"y":470,"wires":[["c9828117.f463"]]},{"id":"61b068dd229b63ee","type":"function","z":"27d4cd61f8c2d5ed","name":"function 2","func":"//{ \"action\": \"bounds\", \"south\": 50.68427770577119, \"west\": -2.4678039550781254, \"north\": 51.41291212935532, \"east\": -0.29113769531250006, \"zoom\": 10 }\nvar p = msg.payload;\n\nmsg.payload = { \"name\": \"zone1\", \"area\": [[p.south,p.west], [p.north,p.east]] }\nreturn msg;","outputs":1,"noerr":0,"initialize":"","finalize":"","libs":[],"x":390,"y":585,"wires":[["a6c23236311ec26b"]]},{"id":"a6c23236311ec26b","type":"delay","z":"27d4cd61f8c2d5ed","name":"","pauseType":"delay","timeout":"5","timeoutUnits":"seconds","rate":"1","nbRateUnits":"1","rateUnits":"second","randomFirst":"1","randomLast":"5","randomUnits":"seconds","drop":false,"allowrate":false,"outputs":1,"x":555,"y":585,"wires":[["c9828117.f463"]]}]
```

**Comment by dceejay** [comment]
Fixed in v2.30.3

**Comment by mrdvt92** [comment]
Testing passed on 2.30.3
